tests/indexes_/test_checks.py

```diff
@@ -0,0 +1,223 @@
+from django_mongodb_backend.fields import ArrayField
+from django_mongodb_backend.indexes import SearchIndex, VectorSearchIndex
+
+
+@skipIfDBFeature("supports_atlas_search")
+@isolate_apps("indexes_", attr_name="apps")
+@override_system_checks([check_vector_search_indexes])
+class InvalidSearchIndexesTests(TestCase):
+    def test_requires_atlas_search_support(self):
+        class Article(models.Model):
+            title = models.CharField(max_length=10)
+
+            class Meta:
+                indexes = [SearchIndex(fields=["title"])]
+
+        errors = checks.run_checks(app_configs=self.apps.get_app_configs(), databases={"default"})
+        self.assertEqual(
+            errors,
+            [
+                checks.Warning(
+                    "This MongoDB server does not support atlas search.",
+                    hint=(
+                        "The index won't be created. Silence this warning if you "
+                        "don't care about it."
+                    ),
+                    obj=Article._meta.indexes[0],
+                    id="django_mongodb_backend.indexes.SearchIndex.W001",
+                )
+            ],
+        )
+
+
+@skipIfDBFeature("supports_atlas_search")
+@isolate_apps("indexes_", attr_name="apps")
+@override_system_checks([check_vector_search_indexes])
+class UnsupportedSearchIndexesTests(TestCase):
+    def test_requires_atlas_search_support(self):
+        class Article(models.Model):
+            title = models.CharField(max_length=10)
+
+            class Meta:
+                indexes = [VectorSearchIndex(fields=["title"])]
+
+        errors = checks.run_checks(app_configs=self.apps.get_app_configs(), databases={"default"})
+        self.assertEqual(
+            errors,
+            [
+                checks.Warning(
+                    "This MongoDB server does not support atlas search.",
+                    hint=(
+                        "The index won't be created. Silence this warning if you "
+                        "don't care about it."
+                    ),
+                    obj=Article._meta.indexes[0],
+                    id="django_mongodb_backend.indexes.VectorSearchIndex.W001",
+                )
+            ],
+        )
+
+
+@skipUnlessDBFeature("supports_atlas_search")
+@isolate_apps("indexes_", attr_name="apps")
+@override_system_checks([check_vector_search_indexes])
+class InvalidVectorSearchIndexesTests(TestCase):
+    def test_requires_size(self):
+        class Article(models.Model):
+            title_embedded = ArrayField(models.FloatField())
+
+            class Meta:
+                indexes = [VectorSearchIndex(fields=["title_embedded"])]
+
+        errors = checks.run_checks(app_configs=self.apps.get_app_configs(), databases={"default"})
+        self.assertEqual(
+            errors,
+            [
+                checks.Error(
+                    "Atlas vector search requires size on title_embedded.",
```

Comment:
You missed a word in my suggestion: "on field"

How about: "An Atlas vector search index requires you to specify Field.size on field Model.field_name."